panel/models/vtk/vtk.ts

```diff
@@ -79,48 +79,42 @@ export class VTKPlotView extends PanelHTMLBoxView {
   }
 
-  after_layout(): void {
-    if (!this._rendererEl) {
-      const container = div({
-        style: {
-          width: "100%",
-          height: "100%"
-        }
-      })
-      this.el.appendChild(container)
-      this._rendererEl = vtkns.FullScreenRenderWindow.newInstance({
-        rootContainer: this.el,
-        container: container
-      })
-      const canvas_list = container.getElementsByTagName('canvas')
-      if(canvas_list.length != 1)
-        throw Error('Error at initialization of the 3D scene, container should have one and only one canvas')
-      else
-        canvas_list[0].classList.add('scene3d-canvas')
-      const axes_canvas = canvas({
-        style: {
-          position: "absolute",
-          top: "0",
-          left: "0",
-          width: "100%",
-          height: "100%"
-        }
-      })
-      axes_canvas.classList.add('axes-canvas')
-      container.appendChild(axes_canvas)
-      this._rendererEl.setResizeCallback(() => {
-        const dims = container.getBoundingClientRect()
-        const width = Math.floor(dims.width * window.devicePixelRatio)
-        const height = Math.floor(dims.height * window.devicePixelRatio)
-        axes_canvas.setAttribute('width', width.toFixed())
-        axes_canvas.setAttribute('height', height.toFixed())
-      })
-      this._plot()
-      this._rendererEl.getRenderer().getActiveCamera().onModified(() => this._get_camera_state())
-      this._remove_default_key_binding()
-      this.model.renderer_el = this._rendererEl
-    }
-    super.after_layout()
+  render(): void {
+    super.render()
+    const container = div()
+    set_size(container, this.model)
+    this.el.appendChild(container)
+    this._rendererEl = vtkns.FullScreenRenderWindow.newInstance({
+      rootContainer: this.el,
+      container: container
+    })
+    const canvas_list = container.getElementsByTagName('canvas')
+    if(canvas_list.length != 1)
+      throw Error('Error at initialization of the 3D scene, container should have one and only one canvas')
+    else
+      canvas_list[0].classList.add('scene3d-canvas')
+    const axes_canvas = canvas({
+      style: {
+        position: "absolute",
+        top: "0",
+        left: "0",
+        width: "100%",
+        height: "100%"
+      }
+    })
+    axes_canvas.classList.add('axes-canvas')
+    container.appendChild(axes_canvas)
+    this._rendererEl.setResizeCallback(() => {
+      const dims = container.getBoundingClientRect()
+      const width = Math.floor(dims.width * window.devicePixelRatio)
+      const height = Math.floor(dims.height * window.devicePixelRatio)
+      axes_canvas.setAttribute('width', width.toFixed())
+      axes_canvas.setAttribute('height', height.toFixed())
+    })
+    this._plot()
+    this._rendererEl.getRenderer().getActiveCamera().onModified(() => this._get_camera_state())
+    this._remove_default_key_binding()
+    this.model.renderer_el = this._rendererEl
```

Comment:
@xavArtley While I'm looking at this code, I'm assuming this is added to make JS linking possible in some form?

Reply:
It's usually bad form to attach anything related to the view to a model but if it's about linking I get why you're doing this.

Reply:
Indeed

Reply:
Thanks, glad I have at least a rudimentary understanding of the VTK models having done this.